src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -2097,4 +2129,265 @@ public boolean equals(Object obj) {
+         */
+        Properties props = System.internalGetProperties();
+        // If the sun.stderr.encoding was already set in System, don't change the encoding
+        if (!System.hasSetErrEncoding()) {
+            Charset stderrCharset = System.getCharset(props.getProperty("sun.stderr.encoding"), true);
+            if (stderrCharset != null) {
+                System.err.flush();
+                System.setErr(System.createConsole(FileDescriptor.err, stderrCharset));
+            }
+        }
+
+        // If the sun.stdout.encoding was already set in System, don't change the encoding
+        if (!System.hasSetOutEncoding()) {
+            Charset stdoutCharset = System.getCharset(props.getProperty("sun.stdout.encoding"), true);
+            if (stdoutCharset != null) {
+                System.out.flush();
+                System.setOut(System.createConsole(FileDescriptor.out, stdoutCharset));
+            }
+        }
+    }
+
+    void uncaughtException(Throwable e) {
+        UncaughtExceptionHandler handler = getUncaughtExceptionHandler();
+        if (handler != null) {
+            handler.uncaughtException(this, e);
+        }
+    }
+
+    /**
+     * Initialize the thread according to its parent Thread and the ThreadGroup where it should be added.
+     *
+     * @param booting Indicates if the JVM is booting up, i.e. if the main thread is being attached
+     * @param threadGroup The ThreadGroup to which the receiver is being added
+     * @param parentThread The creator Thread from which to inherit some values like local storage, etc.
+     *                     If null, the receiver is either the main Thread or a JNI-C attached Thread
+     * @param acc The AccessControlContext. If null, use the current context
+     * @param inheritThreadLocals A boolean indicating whether to inherit initial values for inheritable thread-local variables
+     */
+    private void initialize(boolean booting, ThreadGroup threadGroup, Thread parentThread, AccessControlContext acc, boolean inheritThreadLocals) {
+        tid = nextThreadID();
+        this.group = threadGroup;
+        if (booting) {
+            System.afterClinitInitialization();
+        }
+
+        // initialize the thread local storage before making other calls
+        if (parentThread != null) {
+            // Non-main thread
+            if (inheritThreadLocals && (parentThread.inheritableThreadLocals != null)) {
+                inheritableThreadLocals = ThreadLocal.createInheritedMap(parentThread.inheritableThreadLocals);
+            }
+
+            @SuppressWarnings("removal")
+            final SecurityManager sm = System.getSecurityManager();
+            final Class implClass = getClass();
+            final Class thisClass = Thread.class;
+            if ((sm != null) && (implClass != thisClass)) {
+                boolean override = ((Boolean)AccessController.doPrivileged(new PrivilegedAction() {
+                    public Object run() {
```

Comment:
No raw types, please. This should begin:
```
        boolean override = AccessController.doPrivileged(new PrivilegedAction<Boolean>() {
             @Override
             public Boolean run() {
```
A closing `)` needs to be removed on line 2237.

Reply:
updated.